sklearn/utils/tests/test_validation.py

```diff
@@ -728,9 +744,11 @@ def test_check_dataframe_warns_on_dtype():
     df_mixed_numeric = pd.DataFrame([[1., 2, 3], [4., 5, 6]])
     assert_warns(DataConversionWarning, check_array, df_mixed_numeric,
                  dtype='numeric', warn_on_dtype=True)
-    assert_no_warnings(check_array, df_mixed_numeric.astype(int),
-                       dtype='numeric', warn_on_dtype=True)
-
+    with pytest.warns(DeprecationWarning,
```

Comment:
use pytest.warn(None) and put this in a separate test if you need to ignore the deprecation warning

Reply:
done